Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -675,6 +732,185 @@ void MDNorm::createNormalizationWS(
   }
 }
 
+/**
+ * Validates the TemporaryDataWorkspace has the same binning
+ * as the input binning parameters
+ * @param parameters :: map of binning parameters
+ * @param tempDataWS :: the workspace weare using to aggregate from
+ * @return :: bool - true means the binning is correct to aggreagete using
+ * tempDataWS
+ */
+void MDNorm::validateBinningForTemporaryDataWorkspace(
+    const std::map<std::string, std::string> &parameters,
+    const Mantid::API::IMDHistoWorkspace_sptr tempDataWS) {
+
+  // parse the paramters map and get extents from tempDataWS
+  const std::string numBinsStr = parameters.at("OutputBins");
+  const std::string extentsStr = parameters.at("OutputExtents");
```

Comment:
are these two keys guaranteed to be in `parameters`? Otherwise these functions will throw a potentially confusing `out_of_range` exception.

Reply:
Yes, the `OutputBins` and `OutputExtents` are part of the output from `getBinParameters()`